packages/site/src/components/Connect.tsx

```diff
@@ -0,0 +1,78 @@
+import { ChangeEvent, FormEvent, FunctionComponent, useState } from 'react';
+import { Button, Form } from 'react-bootstrap';
+import { graphql, useStaticQuery } from 'gatsby';
+import { useInstallSnapMutation } from '../api';
+import { ButtonSpinner } from './ButtonSpinner';
+
+interface ConnectProps {
+  name: string;
+  snapId?: string;
+  version?: string;
+}
+
+interface Query {
+  site: {
+    siteMetadata: {
+      version: string;
+    };
+  };
+}
+
+export const Connect: FunctionComponent<ConnectProps> = ({
+  name,
+  snapId: defaultSnapId = '',
+  version: defaultVersion,
+}) => {
+  const [installSnap, { isLoading }] = useInstallSnapMutation();
+  const [snapId, setSnapId] = useState(defaultSnapId);
+  const { site } = useStaticQuery<Query>(graphql`
+    query Version {
+      site {
+        siteMetadata {
+          version
+        }
+      }
+    }
+  `);
+
+  const { version } = site.siteMetadata;
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setSnapId(event.target.value);
+  };
+
+  const handleConnect = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    installSnap({ snapId, version: defaultVersion ?? version });
+  };
+
+  return (
+    <Form onSubmit={handleConnect} className="mb-3">
+      <Form.Group className="mb-3">
+        <Form.Label>Snap ID</Form.Label>
+        <Form.Control
+          type="text"
+          placeholder="Snap ID"
+          value={snapId}
+          onChange={handleChange}
+          data-testid="connect-snap-id"
+          disabled={true}
+        />
+      </Form.Group>
+
+      <Button
+        variant="primary"
+        type="submit"
+        data-testid="connect-button"
+        disabled={isLoading}
```

Comment:
Perhaps we could show something to notify the user that the snap is already installed? It's fine to keep showing the connect button for testing purposes, but it may be nice to be able to determine if it is already installed at a glance.

Reply:
We could change the text to "Reconnect" if it's already installed?

Reply:
That could work 👍